Rx/tst/UTReciever/test_mocks.c

```diff
@@ -0,0 +1,90 @@
+#include <string.h>
+#include <assert.h>
+#include "test_mocks.h"
+
+
+
+int call_ListReaderInit;
+int call_IsListReadingAllowed;
+int MaxCall_IsListReadingAllowed;
+Status_t ret_getbatparamvauluefromlist;
+int call_GetBatteryParamValueFromList;
+int GetBatteryParamValueFromList_value;
+
+void ListReaderInit(void)
+{
+    call_ListReaderInit++;
+}
+
+int IsListReadingAllowed(void)
+{
+    int retval = 1;
+    call_IsListReadingAllowed++;
+    if (call_IsListReadingAllowed ==  MaxCall_IsListReadingAllowed)
+    {
+        retval = 0;
+    }
+    if(ret_getbatparamvauluefromlist == Status_NotOk)
+    {
+        retval = 0;
+    }
+    return retval;
+}
+
+void GenerateIncrementalValues(void)
+{
+    GetBatteryParamValueFromList_value++;
+}
+
+void GenerateDecrementalValues(void)
+{
+    GetBatteryParamValueFromList_value--;
+}
+void (*GenerateValuesForBatteryParam)(void);
+
+Status_t GetBatteryParamValueFromList(int *buf)
+{
+    call_GetBatteryParamValueFromList++;
+    assert(GenerateValuesForBatteryParam!=0);
+    GenerateValuesForBatteryParam();
+    *buf = GetBatteryParamValueFromList_value;
+    return ret_getbatparamvauluefromlist;
+}
+
+char fmt[150];
+char paramname[20];
+int value;
+float MovingAverage;
```

Comment:
Hard to know why this is in global scope and also inside `CalculateMovingAvergaeStateSpecific_t`